src/main/scala/mesosphere/marathon/core/autoscale/impl/MesosAgentCountScalePolicy.scala

```diff
@@ -0,0 +1,63 @@
+package mesosphere.marathon.core.autoscale.impl
+
+import akka.actor.ActorRefFactory
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.api.v2.json.Formats
+import mesosphere.marathon.core.autoscale.{ AutoScaleResult, AutoScalePolicy }
+import mesosphere.marathon.state.{ AutoScalePolicyDefinition, AppDefinition }
+import mesosphere.mesos.{ Constraints, MesosSlaveData }
+import mesosphere.util.ThreadPoolContext
+import mesosphere.util.state.MesosLeaderInfo
+import org.slf4j.LoggerFactory
+import play.api.libs.json.Json
+import spray.client.pipelining._
+import spray.http._
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+class MesosAgentCountScalePolicy(leaderInfo: MesosLeaderInfo, implicit val factory: ActorRefFactory)
+    extends AutoScalePolicy {
+
+  private[this] val log = LoggerFactory.getLogger(getClass)
+  private[this] implicit val ec = ThreadPoolContext.ioContext
+
+  override val name: String = "mesos_agent_count"
+
+  override def scale(definition: AutoScalePolicyDefinition,
+                     app: AppDefinition,
+                     tasks: Seq[MarathonTask]): Future[AutoScaleResult] = {
+    leaderInfo.currentLeaderUrl
+      .map { leaderUrl => slaveData(leaderUrl).map(scaleTo(definition, app, tasks, _)) }
```

Comment:
Please do not retrieve this data for every scaling request but cache it -- at least for a couple of seconds.